src/sentry/static/sentry/app/views/settings/incidentRules/ruleForm/index.tsx

```diff
@@ -80,6 +80,11 @@ class RuleFormContainer extends AsyncComponent<Props, State> {
   getDefaultState(): State {
     const {rule} = this.props;
 
+    // Warning trigger is removed if it is blank when saving
+    if (rule.triggers.length !== 2) {
+      rule.triggers.push(createDefaultTrigger('warning'));
```

Comment:
props needs to be immuitable, mutating props needs to be avoided, this can cause problems e.g. in lifecycle methods when comparing props.